Tests/ForageSDKTests/ForageServiceTests.swift

```diff
@@ -205,6 +205,42 @@ final class ForageServiceTests: XCTestCase {
         }
         wait(for: [expectation], timeout: 1.0)
     }
+
+    func testGetJitterAmountWithCustomRandomGenerator() {
+        let mockSession = URLSessionMock()
+        mockSession.data = forageMocks.capturePaymentSuccess
+        mockSession.response = forageMocks.mockSuccessResponse
+        let service = LiveForageService(provider: Provider(mockSession))
+        
+        let jitterAmount = service.getJitterAmount()
+        
+        // Assert
+        XCTAssertNotNil(jitterAmount, "Jitter amount should not be nil")
+        XCTAssertTrue(jitterAmount >= -0.025 && jitterAmount <= 0.025, "Jitter amount should be within -0.025 and 0.025")
```

Comment:
Normally, we should avoid relying on random values in our unit tests to [keep them deterministic](https://www.browserstack.com/guide/unit-testing-best-practices#:~:text=Ensure%20tests%20are%20deterministic).

Given this is harder to set up and mock in Swift, we can keep the test as-is for now and create a ticket to refactor `getJitterAmount` to be mockable (fix the RNG seed value).

Reply:
Agreed, but technically speaking isn't this test deterministic still? The random value is always going to be between `-0.025` and `0.025`.

Reply:
The random value is always going to be between -0.025 and 0.025.

It's more handy to be able to fix the seed so that there isn't randomness in the functions that use `jitterAmount`, like `waitNextAttempt`. The `expectation` has the right maximum bound, but if we changed the jitter amount range from 25 ms to 300ms, the tests for `waitNextAttempt` could become flaky.

Reply:
we could do something like this to ensure that `waitNextAttempt` has repeatable behavior:

```swift
internal func jitterAmountInSeconds(using rng: () -> Int = defaultRNG) -> Double {
    return Double(rng()) / 1000.0
}

private func defaultRNG() -> Int {
    return Int.random(in: -25...25)
}
```